### PR TITLE
Add calibrate_output_bias option and rename bias_scale to bias_std

### PR DESCRIPTION
**Pull request by shaharbar1**
### Changes:

- Rename bias_scale parameter to bias_std in BaseBayesianNeuralNetwork.cold_start() and BaseBayesianNeuralNetworkMO.cold_start() for naming clarity
- Add calibrate_output_bias field to BaseBayesianNeuralNetwork — fires once on first _update() to set output-layer bias mu to logit(empirical_CVR), preventing over-exploration when the cold-start prior pushes predicted probability toward 0.5
- Add _calibrate_output_bias() method implementing the one-shot calibration, guarded by a bias_calibrated flag that resets on _reset()
- Remove logit clipping in the forward pass (jnp.clip on linear_transform) — calibrated bias makes this unnecessary
- Rename _min_sigma class variable to _numerical_eps to reflect its broader use across epsilon-floor operations
- Update TestBiasScale → TestBiasStd and all related test methods/parameters to match the renamed parameter

## Summary by CodeRabbit

* **New Features**
  * Added optional calibrate_output_bias to initialize output bias from observed rewards; introduces a one-time calibrated flag.

* **Refactor**
  * Renamed public parameter bias_scale → bias_std across cold-start and model APIs; cold-start now accepts calibrate_output_bias.

* **Bug Fixes**
  * Improved numerical stability for posterior standard deviations; final model outputs are no longer clipped.

* **Tests**
  * Updated tests and fixtures; added a binary-reward helper to cover bias_std and calibration.

[Review Change Stack](https://app.coderabbit.ai/change-stack/PlaytikaOSS/pybandits/pull/143?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: pybandits/model.py [hunk 1]
```diff
@@ -1755,7 +1757,7 @@ def model(x: jax.Array, y: jax.Array):
                 # Final output processing
                 logit = numpyro.deterministic(
                     self._logit_var_name,
-                    jnp.clip(linear_transform.squeeze(-1), -_LOGIT_CLIPPING_THRESHOLD, _LOGIT_CLIPPING_THRESHOLD),
```
Comment (galih112) [inline pybandits/model.py:1758]: is it safe to remove it independently? the default to calibrate_output_bias is False

File: pyproject.toml
Comment (galih112) [inline pyproject.toml]: update to recent version

Comment (shaharbar1) [reply pyproject.toml]: For sure I will. I want to rebate on your changes, just didn't have the chance. But the new version would be 7.0.3 anyway

File: tests/test_model.py
Comment (galih112) [inline tests/test_model.py:1]: what about adding new tests for the new functionality of calibrate_output_bias?

File: pybandits/model.py (hunk 1 above)
Comment (shaharbar1) [reply pybandits/model.py:1758]: Yeah, this change is independent of the output bias calibration. Currently, gradients are blocked, this change fixes that.

**Review by galih112 — COMMENTED** [review]
see comments

File: tests/test_model.py
Comment (shaharbar1) [reply tests/test_model.py:1]: I forgot to commit them, will do

File: tests/test_model.py [hunk 2]
```diff
@@ -385,76 +409,201 @@ def test_bias_scale_none_matches_default(
...
+        ],
+    )
+    def test_validator_bias_calibrated_consistency(
+        self,
+        calibrate_output_bias: bool,
+        bias_calibrated: bool,
+        should_raise: bool,
+    ) -> None:
+        """Validator rejects bias_calibrated=True when calibrate_output_bias=False; allows all other combos."""
+        fc = FeaturesConfig(n_features=2)
```
Comment (galih112) [inline tests/test_model.py]: you might want to make n_features a parameter

File: tests/test_model.py [hunk 3]
```diff
@@ -385,76 +409,201 @@ def test_bias_scale_none_matches_default(
...
+        rewards = make_binary_rewards(n_rewards, reward_rate)
+        calibration_bnn._calibrate_output_bias(rewards)
+
+        eps = BaseBayesianNeuralNetwork._numerical_eps
+        empirical_rate = float(np.clip(np.mean(rewards), eps, 1 - eps))
+        expected_intercept = float(np.log(empirical_rate / (1 - empirical_rate)))
+        np.testing.assert_allclose(
+            np.array(calibration_bnn.model_params.bnn_layer_params[-1].bias.params["mu"]),
+            expected_intercept,
+            rtol=1e-5,
```
Comment (galih112) [inline tests/test_model.py]: same for rtol?

**Review by galih112 — APPROVED** [review]
approved

File: tests/test_model.py (hunk 3 above)
Comment (shaharbar1) [reply tests/test_model.py]: Thx!
